_posts/en/newsletters/2019-02-12-newsletter.md

```diff
@@ -0,0 +1,164 @@
+  report provides visual examples of how many systems fail to handle RBF
+  correctly.  This is done not to criticize the pioneering developers of
+  those systems, but to help all Bitcoin developers learn how to master
+  the powerful fee-management capability that RBF provides.  The report
+  also highlights several RBF-related usability enhancements some
+  software have adopted and that other programs may want to consider
+  using for themselves.  Based on the examples collected, the report
+  concludes with a summary of recommendations for developers.
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][bitcoin core repo], [LND][lnd repo], [C-Lightning][c-lightning repo],
+[Eclair][eclair repo], and [libsecp256k1][libsecp256k1 repo].*
+
+- [Bitcoin Core #14897][] introduces a semi-random order biased towards
+  outbound connections when requesting transactions, making it harder
+  for attackers to abuse one of Bitcoin Core's bandwidth-reduction
+  measures.  Previously, when your node received an announcement of a
+  new transaction from one of its peers, it requested that transaction
+  from that peer.  While it waited for the transaction to be sent, it
+  might have received announcements of the same transaction from its
+  other peers.  If the first peer hadn't sent the transaction within two
+  minutes, your node would then request the transaction from the second
+  peer who announced it, again waiting two minutes before requesting it
+  from the next peer.  This allowed an attacker who opens a large number
+  of connections to your node to potentially delay your receipt of a
+  transaction by a large number of two-minute intervals.
+
+    If such an attack was performed across the whole network, it might
+    be able to prevent certain transactions from reaching miners,
+    possibly breaking the security of protocols that rely on timely
+    confirmation (e.g. LN payment channels).  A network-wide attack
+    could also make it [easier][coinscope] to [map][txprobe] the network
+    and redirect transaction traffic in order to learn which IP address
+    originated a transaction.
+
+    With this PR, your node will only immediately request the
+    transaction from the first peer that announced it if your node
+    initially chose to open a connection to that peer (i.e. an outbound
+    peer).  If you first heard about the transaction from a peer that
+    connected to you (an inbound peer), you'll wait two seconds before
+    requesting the transaction to give an outbound peer a chance to tell
+    you about it first.  If the first peer you request the transaction
+    from has not sent it to you within a minute, you'll request it from
+    another randomly-selected peer.  If that also doesn't work, you'll
+    continue to randomly select peers to request the transaction from.
+    This doesn't completely eliminate the problem, but it does mean that
```

Comment:
Nit: Eliminate imply complete. Could write "eliminate" instead of "completely eliminate".